pkg/function/examples/even_odd_key/main.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	functionsdk "github.com/numaproj/numaflow-go/pkg/function"
+	"github.com/numaproj/numaflow-go/pkg/function/server"
+)
+
+func handle(_ context.Context, key string, d functionsdk.Datum) functionsdk.Messages {
+	msg := d.Value()
+	_ = d.EventTime() // Event time is available
+	_ = d.Watermark() // Watermark is available
+	// If msg is not an integer, drop it, otherwise return it with "even" or "odd" key.
+	if num, err := strconv.Atoi(string(key)); err != nil {
```

Comment:
Let me know if it doesn't make sense but the idea is to have a mechanism to aggregate by key, not by value. In Redis Streams, say somebody publishes:
- key 1, value 1
- key 4, value 1
- key 2, value 1
The idea was that you might want to aggregate all of the odd keys together and all of the even keys together.

Reply:
Admittedly, I know it's a bit odd to have a Key that's an integer. It was just supposed to serve as an example of taking incoming keys and grouping them into new keys. If you think it doesn't make sense let me know. Also, we definitely don't need to incorporate it into here since I don't think anyone would ever actually use it.

Reply:
Usually, the cardinality of the key is limited. Most of the operations are on the Value. The Key is derived from the Value.

Reply:
Yes, that's true as far as the cardinality of the key. I am curious if there's another scenario in which there is a source of data that is either Kafka or Redis Streams in which the incoming Keys are not derived from the Value, where it might make sense to have a UDF that takes those incoming keys and groups them into new keys?

Reply:
Yes, very possible. It could be set by the producer when writing to the stream.